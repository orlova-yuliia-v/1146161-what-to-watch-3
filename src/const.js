export const ALL_GENRES = `All genres`;

export const DEFAULT_SHOWED_MOVIES_NUMBER = 8;

export const MAX_SIMILAR_MOVIES_NUMBER = 4;

export const MAX_GENRES_NUMBER = 10;

export const ReviewLength = {
  MIN: 50,
  MAX: 400
};

export const TabName = {
  OVERVIEW: `overview`,
  DETAILS: `details`,
  REVIEWS: `reviews`
};

export const AppRoute = {
  LOGIN: `/login`,
  ROOT: `/`,
  ADD_REVIEW: `/review`,
  MY_LIST: `/mylist`,
  FILMS: `/films`,
  PLAYER: `/player`
};
