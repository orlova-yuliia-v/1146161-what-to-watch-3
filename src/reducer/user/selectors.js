import NameSpace from "../name-space.js";

export const getAuthorizationStatus = (state) => {
  return state[NameSpace.USER].authorizationStatus;
};

export const getAuthUser = (state) => {
  return state[NameSpace.USER].authUserData;
};
