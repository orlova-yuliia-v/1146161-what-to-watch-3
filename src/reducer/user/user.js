import {extend, normalizeUserData} from "../../utils.js";

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`
};

const initialState = {
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  authUserData: {}
};

const ActionType = {
  REQUIRE_AUTHORIZATION: `REQUIRE_AUTHORIZATION`,
  SET_AUTH_USER_DATA: `SET_AUTH_USER_DATA`
};

const Operation = {
  checkAuth: () => (dispatch, _getState, api) => {

    return api
    .get(`/login`)
      .then((response) => {
        if (!response) {
          return;
        }
        const normalizedAuthData = normalizeUserData(response.data);
        dispatch(ActionCreator.setAuthUserData(normalizedAuthData));
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
      })

      .catch((err) => {
        throw err;
      });
  },
  login: (authData) => (dispatch, _getState, api) => {
    return api
      .post(`/login`, {
        email: authData.login,
        password: authData.password
      })
      .then((response) => {
        if (!response) {
          return;
        }
        const normalizedAuthData = normalizeUserData(response.data);
        dispatch(ActionCreator.setAuthUserData(normalizedAuthData));
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH));
      })

      .catch((err) => {
        throw err;
      });
  }
};

const ActionCreator = {
  requireAuthorization: (status) => {
    return {
      type: ActionType.REQUIRE_AUTHORIZATION,
      payload: status
    };
  },
  setAuthUserData: (authUserData) => {
    return {
      type: ActionType.SET_AUTH_USER_DATA,
      payload: authUserData
    };
  }
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.REQUIRE_AUTHORIZATION:
      return extend(state, {
        authorizationStatus: action.payload
      });
    case ActionType.SET_AUTH_USER_DATA:
      return extend(state, {
        authUserData: action.payload
      });
  }

  return state;
};

export {reducer, ActionCreator, ActionType, AuthorizationStatus, Operation};
