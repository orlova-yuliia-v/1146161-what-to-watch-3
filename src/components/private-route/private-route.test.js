import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import PrivateRoute from "./private-route.jsx";
import configureStore from "redux-mock-store";
import {AuthorizationStatus} from "../../reducer/user/user.js";
import {BrowserRouter} from "react-router-dom";
import {AppRoute} from "../../const.js";
import NameSpace from "../../reducer/name-space.js";

const mockStore = configureStore([]);

const authDataMock = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver.conner`,
  avatarUrl: `img/1.png`
};

it(`PrivateRoute is renderer correctly`, () => {
  const store = mockStore({
    [NameSpace.USER]: {
      authorizationStatus: AuthorizationStatus.NO_AUTH,
      authUserData: {authDataMock}
    }
  });
  const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <PrivateRoute render={() => (<div>DIV</div>)} path={AppRoute.LOGIN} exact={true} authorizationStatus={AuthorizationStatus.AUTH} />
        </BrowserRouter>
      </Provider>, {
        createNodeMock: () => {
          return {};
        }
      }
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
