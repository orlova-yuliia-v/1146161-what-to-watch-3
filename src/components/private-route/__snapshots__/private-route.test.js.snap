// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute is renderer correctly 1`] = `null`;
